### PR TITLE
Add initial e2e tests

### PR DESCRIPTION
**Pull request by maxcnunes**
Add initial e2e tests. Although, for more advanced testing coverage we probably will need to upgrade Electron so it gets a recent Chromium version which Playwright depends on, for instance `click` wasn't working because of that, I will try to do that in my next PR. I know SQLite doesn't support the latest version of Electron, but I think if we stop using remote, keep all the DB calls in the browser, and use IPC events only for the renderer integration it probably should work (example https://github.com/sqlectron/sqlectron-gui/blob/656d868e1ed64829ea0fe5b6e01a38167d3eae37/src/browser/preload.ts). 

Anyway, that is a subject for the next PR, for now, this initial support should help us start improving the test coverage. We also will need unit tests, but I think that is gonna be simpler to do once we are not using the remote in the renderer anymore, so it is completely decoupled of the electron API.

### CODE REVIEW

**Comment by MasterOdin** [comment]
Do you know the minimal version of Electron we would need to upgrade to support what you need?

**Comment by maxcnunes** [comment]
By this report https://github.com/microsoft/playwright/issues/1042 it is broken on Chromium 80 and it should work on Chromium 82. Our current Electron version 8 is on Chromium 80 and the next version 9 is on Chromium 83. So, going to Electron 9 should work, although it would be awesome if we could go all over to the latest version.

**Comment by MasterOdin** [comment]
Alright. I think moving to Electron 9 shouldn't be too much of an issue, but going to the latest might given that sqlite3 library is only tested up to Electron 8 and I believe one of the versions between 9 and latest changed how native extensions work.

**Comment by maxcnunes** [comment]
Sure, that is why I mentioned that approach I would like to test. In case that doesn't work then I will try version 9.

**Comment by maxcnunes** [comment]
@MasterOdin I got the e2e testing working for Sqlite basic operations:
* Connect to Sqlite DB
* Assert tables list
* Assert default select query
* Assert rows response (this one for some reason isn't working on the CI, I left it disabled for now on the CI)

I used another API method of playwright to avoid the issue I described before, so it wasn't required to upgrade electron because of that.

File: test/e2e/test.sqlite.js
Comment (MasterOdin) [inline test/e2e/test.sqlite.js:103]:
```suggestion
      expect(await rows[0].innerText()).to.be.equal('info');
      for (let i = 1; i < rows.length; i++) {
        await expect(await rows[i].innerText()).to.be.equal(`Ipsum ${i - 1}`);
      }
```

File: test/e2e/test.sqlite.js
Comment (MasterOdin) [inline test/e2e/test.sqlite.js:26]:
```suggestion
  const db = new sqlite3.Database(DB_PATH);
```

File: test/e2e/test.sqlite.js
Comment (MasterOdin) [inline test/e2e/test.sqlite.js:31]:
```suggestion
    const stmt = db.prepare('INSERT INTO document VALUES (?)');
```

File: test/e2e/test.sqlite.js
Comment (MasterOdin) [inline test/e2e/test.sqlite.js:32]:
```suggestion
    for (let i = 0; i < 10; i++) {
```

File: src/renderer/components/database-item.jsx
Comment (MasterOdin) [inline src/renderer/components/database-item.jsx:174]: I like having a divider here between the class and the metatype, similar to how other libraries do it, eg. `item__${dbObjectType}` or `item-${dbObjectType}`.

**Review by MasterOdin — APPROVED** [review]
Some small changes, otherwise I think it looks good

**Comment by MasterOdin** [comment]
With regards to electron and sqlite3, I would somewhat hold off on doing any of that. I am potentially leaning towards removing node-sqlite3 for better-sqlite3 depending on getting duplicate column results to work. I know that node-sqlite3 does not, and given the level of maintainer support there, I'm not confident it would anytime soon, so seriously considering better-sqlite3, in which case it might make easier the electron upgrade stuff.

**Comment by maxcnunes** [comment]
@MasterOdin Alright, I can hold off that upgrade. Although, I probably will go ahead and prepare the code to support recent Electron version which is not using remote in the renderer and use direct IPC calls instead ([example](https://github.com/sqlectron/sqlectron-gui/blob/656d868e1ed64829ea0fe5b6e01a38167d3eae37/src/browser/preload.ts)).

**Pull request merged**
